### ISSUE TITLE
Deprecate the processors module option

### ISSUE THREAD

**Issue by fwoelffel** [fwoelffel/nest-bull#95]
The `BullModuleOptions.processors` property has become redundant since the implementation of the decorators which provide a more elegant solution.
I should deprecate this ASAP and remove this property in a future release to reduce the code complexity.

**Comment by victordidenko** [comment]
@fwoelffel Before removing `processors` property, please, provide an example, how to run job handlers in separate processes (using file name)

**Comment by fwoelffel** [comment]
Oopsie, I totally forgot about that feature. I'll come back to you with a proper implementation / example.

**Comment by igor-sky** [comment]
@fwoelffel Is example available anywhere? Completely stuck on this one, can't sell **nest-bull** to lead without showing how to use workers in separate process :(

**Comment by fwoelffel** [comment]
Ok. I've had some time to think about this and I've come to the conclusion that the `processors` option should not be deprecated.

However, this options' type will change from
```ts
export type BullQueueProcessor =
  BullQueueProcessorCallback // (job: Job, done: DoneCallback) => void
  | BullQueueAdvancedProcessor // { concurrency?: number; name?: string; callback: BullQueueProcessorCallback; }
  | BullQueueSeparateProcessor // string
  | BullQueueAdvancedSeparateProcessor; // { concurrency?: number; name?: string; path: string; }
```

to

```ts
export type BullQueueProcessor =
  BullQueueSeparateProcessor // string
  | BullQueueAdvancedSeparateProcessor; // { concurrency?: number; name?: string; path: string; }
```

@igor-sky there's already an example in the [README](https://github.com/fwoelffel/nest-bull#separate-processes). Doesn't this work for you?

**Comment by fwoelffel** [comment]
Closed by 1dd0303aaef8dfbf09268af0521dcffcd92e1338
